src/js/tabs/osd.js

```diff
@@ -1728,6 +1878,18 @@ TABS.osd.initialize = function (callback) {
             }
         });
 
+        // replace logo
+        $('a.replace_logo').click(function () {
+            if (!GUI.connect_lock) { // button disabled while flashing is in progress
+              openLogoImage().then(function(ctx) {
+                FONT.replaceLogoFromImage(ctx);
+                FONT.logoPreview($logoPreview);
+              }).catch(function(error) {
+                console.error("error loading image:", error);
```

Comment:
This error appears in other place to the user? If not it will be a good idea to show it in the GUI log too, something like:
`GUI.log(i18n.getMessage('logoLoadingImageError', error));`
or similar.

Reply:
Good catch! I've added internationalized messages to be shown on the GUI.